c2rust-ast-builder/src/builder.rs

```diff
@@ -330,97 +277,29 @@ impl Make<TokenStream> for Vec<TokenTree> {
     }
 }
 
-impl Make<TokenStream> for Vec<String> {
-    fn make(self, _mk: &Builder) -> TokenStream {
-        let mut tokens = vec![];
-        let mut first = true;
-
-        for s in self {
-            if !first {
-                tokens.push(TokenTree::Punct(Punct::new(',', Spacing::Alone)));
-            } else {
-                first = false;
-            }
-
-            tokens.push(TokenTree::Ident(Ident::new(&s, Span::call_site())));
-        }
-
-        tokens.into_iter().collect::<TokenStream>()
-    }
-}
-
 impl Make<TokenStream> for Vec<&str> {
     fn make(self, _mk: &Builder) -> TokenStream {
-        let mut tokens = vec![];
-        let mut first = true;
-
-        for s in self {
-            if !first {
-                tokens.push(TokenTree::Punct(Punct::new(',', Spacing::Alone)));
-            } else {
-                first = false;
-            }
-
-            tokens.push(TokenTree::Ident(Ident::new(s, Span::call_site())));
-        }
-
-        tokens.into_iter().collect::<TokenStream>()
+        comma_separated(self.iter().map(|&s| Ident::new(s, Span::call_site())))
     }
 }
 
 impl Make<TokenStream> for Vec<u64> {
     fn make(self, _mk: &Builder) -> TokenStream {
-        let mut tokens = vec![];
-        let mut first = true;
-
-        for s in self {
-            if !first {
-                tokens.push(TokenTree::Punct(Punct::new(',', Spacing::Alone)));
-            } else {
-                first = false;
-            }
-
-            tokens.push(TokenTree::Literal(Literal::u64_unsuffixed(s)));
-        }
-
-        tokens.into_iter().collect::<TokenStream>()
+        comma_separated(self.iter().map(|&s| Literal::u64_unsuffixed(s)))
     }
 }
 
 impl Make<TokenStream> for Vec<Meta> {
     fn make(self, _mk: &Builder) -> TokenStream {
-        let mut tokens = TokenStream::new();
-
-        let mut first = true;
-
-        for meta in self {
-            if !first {
-                let tt = TokenTree::Punct(Punct::new(',', Spacing::Alone));
-
-                tokens.extend(vec![tt]);
-            } else {
-                first = false;
-            }
-
-            meta.to_tokens(&mut tokens);
-        }
-
-        tokens
+        comma_separated(self.iter())
     }
 }
-
```

Comment:
How come this newline is deleted?

Reply:
hmm, maybe some `rebase` shenanigan😅. I will fix it now